File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/ForegroundToBackgroundTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class ForegroundToBackgroundTransformer extends ABaseTransformer {

	@Override
	protected void onTransform(final View view, final float position) {
		final float height = view.getHeight();
		final float width = view.getWidth();
		final float scale = min(position > 0 ? 1F : Math.abs(1F + position), 0.5F);
		view.setScaleX(scale);
		view.setScaleY(scale);
		view.setPivotX(width * 0.5F);
		view.setPivotY(height * 0.5F);
		view.setTranslationX(position > 0 ? width * position : -width * position * 0.25F);
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/ForegroundToBackgroundTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class ForegroundToBackgroundTransformer : ABaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        val height = page.height.toFloat()
        val width = page.width.toFloat()
        val scale = min(if (position > 0) 1.toFloat() else Math.abs(1.toFloat() + position), 0.1.toFloat())
        page.scaleX = scale
        page.scaleY = scale
        page.pivotX = width * 0.1.toFloat()
        page.pivotY = height * 0.1.toFloat()
        page.translationX = if (position > 0) width * position else -width * position * 0.1.toFloat()
    }

}
